### ISSUE TITLE
Oracle UPDATE .. RETURNING emulation doesn't work for boolean data type rewrites

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11749]
As reported by a customer, when using `BOOLEAN` types (via forced types) with `UPDATE .. RETURNING` statements, we currently cannot fetch the resulting values:

```
java.lang.IllegalArgumentException: Data Type cannot be mapped to DBMS_SQL.XXX_TABLE type: boolean (java.lang.Boolean)
      at org.jooq_3.13.4.ORACLE.debug(Unknown Source)
      at org.jooq.impl.AbstractDMLQuery.dbmsSQLType(AbstractDMLQuery.java:680)
      at org.jooq.impl.AbstractDMLQuery.accept(AbstractDMLQuery.java:433)
      at org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:591)
      at org.jooq.impl.AbstractContext.visit0(AbstractContext.java:457)
      at org.jooq.impl.AbstractContext.visit(AbstractContext.java:218)
      at org.jooq.impl.AbstractQuery.getSQL0(AbstractQuery.java:523)
      at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:326)
      at org.jooq.impl.UpdateImpl.fetchOne(UpdateImpl.java:872)
```

This issue is still present in 3.15.0-SNAPSHOT. The fix should be simple - we can just use numbers and delay the conversion to booleans until later.

**Comment by lukaseder** [comment]
The problem at the customer site is that they're using data type rewriting to force `NUMBER(1)` containing `1` or `0`, or `CHAR(1)` types containing `'1'` or `'0'` to `BOOLEAN`, rather than using `Converter<T, Boolean>` or `Binding<T, Boolean>` on them. In the latter case, jOOQ would correctly use the `<T>` type to choose `DBMS_SQL.NUMBER_TABLE` or `DBMS_SQL.VARCHAR2_TABLE` behind the scenes.

We could nevertheless fix this and try to map `BOOLEAN` to `DBMS_SQL.NUMBER_TABLE`, as this seems to be the most reasonable choice here.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.9 (#11776)
